### PR TITLE
Runtime version tag refactor

### PR DESCRIPTION
**Pull request by MichaelSimons**
@richlander, @glennc, @naamunds, @dleeapho 

~~This is a preview of the refactored Dockerfiles for the runtime focused tagging.  I still need to update CI tests and the update-dependencies logic.  Would appreciate any feedback you may have.  The tag links in README currently point to my branch if you would like to use it to browse the Dockerfiles.~~

Refactor the nightly repo so that it is a true staging environment for the official repo.  This includes using the tags that will appear in the official repo once released.  This repo will no longer align with CLI channels.  As part of this re-work, the sdk tagging strategy is changing.  The sdk tag version will align the the runtime instead of the cli.  This is intended to identify the sdk image users should use for a particular runtime.  The goal is to eliminate the confusion with the different runtime and sdk versions that existed previously.  Additionally there are two sdk variants for the interim - project json and msbuild.  There will only be one in the long run but until the port to msbuild is complete we will have two flavors.

Updating the update-dependencies logic is out of scope for this PR.  This will be handled with a separate PR.

### CODE REVIEW

File: 1.1/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+You can find samples, documentation, and getting started instructions for .NET Core in the [.NET Core documentation](https://docs.microsoft.com/dotnet/articles/core/).
+
+[![Downloads from Docker Hub](https://img.shields.io/docker/pulls/microsoft/dotnet-nightly.svg)](https://hub.docker.com/r/microsoft/dotnet-nightly)
+[![Stars on Docker Hub](https://img.shields.io/docker/stars/microsoft/dotnet-nightly.svg)](https://hub.docker.com/r/microsoft/dotnet-nightly)
+
+
+## Supported tags
+
+-       ['1.0.2-runtime', '1.0-runtime' (*1.0/debian/runtime/Dockerfile*)](https://github.com/MichaelSimons/dotnet-docker-nightly/blob/runtime-version-tag-refactor/1.0/debian/runtime/Dockerfile)
```
Comment (richlander) [inline 1.1/README.md]: > https://github.com/MichaelSimons/

I was expecting these to be https://github.com/dotnet/

**Comment by richlander** [comment]
LGTM

**Comment by dleeapho** [comment]
:shipit:

**Comment by naamunds** [comment]
It looks like "staging" is misspelled in the commit name.

File: test/run-test.sh [hunk 2]
```diff
@@ -6,46 +6,30 @@ docker_repo="microsoft/dotnet-nightly"
 repo_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/.."
 
 # Maps development image versions to the corresponding runtime image version
```
Comment (MichaelSimons) [inline test/run-test.sh]: Fixed.

File: build-and-test.ps1 [hunk 3]
```diff
@@ -6,7 +6,7 @@ $dockerRepo="microsoft/dotnet-nightly"
 pushd $PSScriptRoot
 
 Get-ChildItem -Recurse -Filter Dockerfile | where {$_.DirectoryName.TrimStart($PSScriptRoot) -like "*\nanoserver*"} | sort DirectoryName | foreach {
-    $tag = "$($dockerRepo):" + $_.DirectoryName.Replace($PSScriptRoot, '').TrimStart('\').Replace('\', '-')
+    $tag = "$($dockerRepo):" + $_.DirectoryName.Replace($PSScriptRoot, '').Replace("\nanoserver", '').TrimStart('\').Replace('\', '-') + "-nanoserver"
```
Comment (MichaelSimons) [inline build-and-test.ps1]: I go back and forth on this topic.  I don't see a real benefit either way.  This is the pattern we used before so I kept it.  The nice thing about this pattern is that you can go to the repo and easily browse by platform.  Not making any changes at this time.

File: 1.1/nanoserver/sdk/msbuild/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+# Trigger the population of the local package cache
+ENV NUGET_XMLDOC_MODE skip
+RUN mkdir warmup \
+    && cd warmup \
+    && dotnet new \
+    && cd .. \
+    && rmdir /q/s warmup
+
+
```
Comment (MichaelSimons) [inline 1.1/nanoserver/sdk/msbuild/Dockerfile]: removed

File: 1.0/debian/sdk/msbuild/Dockerfile [hunk 5]
```diff
@@ -16,15 +16,23 @@ RUN apt-get update \
...
+    && rm dotnet.tar.gz \
+    && ln -s /usr/share/dotnet/dotnet /usr/bin/dotnet
+
 # Install .NET Core SDK
 ENV DOTNET_SDK_VERSION 1.0.0-preview3-003933
 ENV DOTNET_SDK_DOWNLOAD_URL https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$DOTNET_SDK_VERSION/dotnet-dev-debian-x64.$DOTNET_SDK_VERSION.tar.gz
 
 RUN curl -SL $DOTNET_SDK_DOWNLOAD_URL --output dotnet.tar.gz \
-    && mkdir -p /usr/share/dotnet \
     && tar -zxf dotnet.tar.gz -C /usr/share/dotnet \
```
Comment (MichaelSimons) [inline 1.0/debian/sdk/msbuild/Dockerfile]: yes it overwrites the files.  We are explicitly using the tools version of the muxer.

**Pull request merged**